### DISCUSSION TITLE
1 to many and many to 1 connections

### DISCUSSION THREAD

**Discussion by amitjoy** [eckig/graph-editor#98] [Q&A] [answered]
Could you kindly tell me how to create connections between nodes in 1 to many (1 output connector to multiple input connectors) and many to 1 (many output connectors to 1 input connector) relationships? As far as I have seen, these are not possible. Maybe I am missing something. 

Very eagerly looking forward to your further help.

**Comment by eckig** [comment, marked as answer]
For both modes you need to supply your own `io.github.eckig.grapheditor.GConnectorValidator` that allows multiple connections from and/or to connectors.

This is a quick hack that does the trick, so you know where to start:
```
graphEditor.setConnectorValidator(new DefaultConnectorValidator(){
            @Override
            public boolean validate(final GConnector source, final GConnector target) {

                if (source.getType() == null || target.getType() == null) {
                    return false;
                } else if (source.getParent().equals(target.getParent())) {
                    return false;
                }

                final boolean sourceIsInput = DefaultConnectorTypes.isInput(source.getType());
                final boolean targetIsInput = DefaultConnectorTypes.isInput(target.getType());

                return sourceIsInput != targetIsInput;
            }
        });
```

By default, if a connector is already connected, trying to create a new connection from that connector "only" detaches the existing connection. To disable this behavior you need to set `connector.setConnectionDetachedOnDrag(false);` on every `GConnector` where you want this behavior.

**Reply by amitjoy** [reply]
Thanks a lot for your quick answer 👍
